tests/test_fixed.py

```diff
@@ -0,0 +1,85 @@
+    raw_a, event_a,_ = simulations.simulate(sources, n_trials, cpus, 'dataset_a_raw', overwrite=True,
+                                          sfreq=sfreq, times=times_a, noise=True, seed=1, save_snr=True)
+    means = np.array([50, 100, 100, 50])/2
+    sources = []
+    for cur_name, cur_mean in zip(names, means):
+        sources.append([cur_name, 10., 1, gamma(2, scale=cur_mean)])
+    raw_b, event_b = simulations.simulate(sources, n_trials, cpus, 'dataset_b_raw', seed=1, overwrite=True, 
+        sfreq=sfreq, verbose=True, proportions=[.99,1,1,1], noise=False)
+    
+    events = []
+    raws = []
+    event_id = {'stimulus':1}#trigger 1 = stimulus
+    resp_id = {'response':5}
+    
+    for raw_file, event_file in zip([raw_a, raw_b],
+                    [event_a, event_b]):
+        events.append(np.load(event_file))
+        raws.append(raw_file)
+    sfreq = 100
+    return raws, events, event_id, resp_id
+
+def test_fixed():
+    sfreq = 100
+    n_events = 3
+    raws, events, event_id, resp_id = create_data(sfreq, n_events)
+    event_a = events[0]
+    # Data reading
+    epoch_data = hmp.utils.read_mne_data(raws, event_id=event_id, resp_id=resp_id, sfreq=sfreq,
+            events_provided=events, verbose=True, reference='average', high_pass=1, low_pass=45,
+                subj_idx=['a','b'],pick_channels='eeg', lower_limit_rt=0.01, upper_limit_rt=2 ) 
+    epoch_data = epoch_data.assign_coords({'condition': ('participant', epoch_data.participant.data)})
+    positions = simulations.simulation_positions()
+
+    hmp_data = hmp.utils.transform_data(epoch_data, n_comp=2,)
+
+        
+   # Comparing to simulated data, asserting that results are the one simulated
+    data_a = hmp.utils.participant_selection(hmp_data, 'a')
+    event_properties = EventProperties.create_expected(sfreq=data_a.sfreq)
+    trial_data = TrialData.from_standard_data(data=data_a, template=event_properties.template)
+    init_sim = FixedEventModel(event_properties, n_events=n_events, maximization=False,)
+    sim_source_times, true_pars, true_magnitudes, _ = \
+        simulations.simulated_times_and_parameters(event_a, init_sim, trial_data)
+    true_loglikelihood, true_estimates = init_sim.fit_transform(trial_data, parameters = np.array([true_pars]), magnitudes=np.array([true_magnitudes]),  verbose=True)
+    true_topos = hmp.utils.event_topo(epoch_data, true_estimates.squeeze(), mean=True)
+    init_sim = FixedEventModel(event_properties, n_events=n_events, maximization=True,)
+    likelihood, estimates = init_sim.fit_transform(trial_data, verbose=True)
+    test_topos = hmp.utils.event_topo(epoch_data, estimates.squeeze(), mean=True)
+    assert (np.array(simulations.classification_true(true_topos,test_topos)) == np.array(([0,1,2],[0,1,2]))).all()
+    print(np.sum(np.abs(true_topos.data - test_topos.data)))
+    assert np.round(np.sum(np.abs(true_topos.data - test_topos.data)), 4) == 0
+    assert np.round(likelihood,4) == np.array(2.4834)
```

Comment:
For this purpose, it is nice to use `np.isclose`. You can set the relative and absolute tolerance.

Reply:
I made a commit for that